### PR TITLE
MAID-2373: separate app-specific reducer cases out of initialiser

### PR DESCRIPTION
**Pull request by hunterlester**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @ustulation

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by joshuef** [comment]
Just looking this over it seems pretty good.

Couple small things:

- I don't think we need separate folders for reducers in the email app, just `initialiser.js` and `mail.js` would be fine.
- I think we should have a separate `safe_comms.js` for the initialiser.
- Let's separate out the initialiser actionTypes too/

With these things set up, we could then move the initialiser files to it's own example folder in the repo (maybe `js_authenticator_example` or something?), and then symlink those files back into the example (assuming that this works with packaging... it might not).

And then we're set up to do the implement this for _webhosting_

**Comment by hunterlester** [comment]
@joshuef Things are looking better, though I need help with symlinks.
I created a symlink:
symlink_created

But when a file tries to import the module to which the symlink points:
cannot_find_module_symlink

**Comment by joshuef** [comment]
@hunterlester let's change `example_common_app_utils` to `example_common_js_app_utils` just for clarity (as we've java in the example repo also).

I'm having problems running in prod mode. Seems to keep trying to run the `testAuth` functionality.

https://www.dropbox.com/s/iqloe6w9zn7ky5n/Screenshot%202017-10-04%2014.08.50.jpg?dl=0

File: email_app/app/index.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const createWindow = () => {
   mainWindow.loadURL(`file://${__dirname}/app.html`);
 
   // Open the DevTools.
-  // mainWindow.webContents.openDevTools();
+  mainWindow.webContents.openDevTools();
```
Comment (bochaco) [inline email_app/app/index.js]: Apart from this, everything else works fine on Linux.

File: email_app/app/actions/initialiser_actions.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      type: ACTION_TYPES.NET_STATUS_CHANGED,
+      payload: state
+    });
+  }
+};
+
+export const receiveResponse = (uri) => {
+  return function (dispatch, getState) {
+    let appStatus = getState().emailApp.appStatus;
+    console.log('receiveResponse appStatus: ', appStatus);
```
Comment (joshuef) [inline email_app/app/actions/initialiser_actions.js]: let's remove these comments

Comment (joshuef) [reply email_app/app/actions/initialiser_actions.js]: also line `31`, `37` and `66` here

**Pull request merged**